### PR TITLE
zdtm: fix zdtm/static/maps00 case in arm64

### PR DESCRIPTION
**Pull request by time-river**
zdtm/static/maps00 case sometimes will cause SIGILL signal in arm64 platform.
<<ARM Coretex-A series Programmer's Guide for ARMv8-A>> notes:
> The ARM architecture does not require the hardware to ensure coherency
> between instruction caches and memory, even for locations of shared
> memory.

Therefore, we need flush dcache and icache for self-modifying code.

### CODE REVIEW

**Comment by adrianreber** [comment]
> The python2 is deprecated, switch python3, add src encodings.

Thanks for the pull request, but we want to support CentOS/RHEL 7 until the end of its lifetime which is 2024. CentOS/RHEL 7 has a working and supported Python 2. Until then we want to keep Python 2 support (if possible).

Using an unversioned Python binary like in `zdtm.py` is not really a sign of Python 2. Looking at https://www.python.org/dev/peps/pep-0394/ the version the `python` binary can either point to Python 2 or Python 3.

**Comment by time-river** [comment]
> > The python2 is deprecated, switch python3, add src encodings.
> 
> Thanks for the pull request, but we want to support CentOS/RHEL 7 until the end of its lifetime which is 2024. CentOS/RHEL 7 has a working and supported Python 2. Until then we want to keep Python 2 support (if possible).
> 
> Using an unversioned Python binary like in `zdtm.py` is not really a sign of Python 2. Looking at https://www.python.org/dev/peps/pep-0394/ the version the `python` binary can either point to Python 2 or Python 3.

All right. I drop this commit.

**Comment by adrianreber** [comment]
Please rebase to get the CI errors fixed.

File: test/zdtm/static/maps00.c [hunk 1]
```diff
@@ -158,7 +158,8 @@ static int check_map(struct map *map)
 
 	if (!sigsetjmp(segv_ret, 1)) {
 		if (map->prot & PROT_WRITE) {
-			memcpy(map->ptr, test_func, getpagesize());
+			memcpy(map->ptr, test_func, ONE_MAP_SIZE);
```
Comment (adrianreber) [inline test/zdtm/static/maps00.c:161]: Why the change from `getpagesize()` to `ONE_MAP_SIZE`?

Comment (time-river) [reply test/zdtm/static/maps00.c:161]: I find `mmap(ONE_MAP_SIZE)` instead of `mmap(getpagesize())`. `getpagesize() == ONE_MAP_SIZE` isn't always correct, changing it to prevent the potential overflow.

Comment (adrianreber) [reply test/zdtm/static/maps00.c:161]: Looking at the code of the test I agree. It is strange that `getpagesize()` is used here.

It does not really fit the commit description, but splitting in two one-line commits seems unnecessary. I am okay with this.

**Review by adrianreber — APPROVED** [review]
CI is happy. The description sounds reasonable. I have never seen this error but I am not running CRIU very often on aarch64.

**Comment by adrianreber** [comment]
@Snorch this PR looks good from my point of view. Can you merge it if it also makes sense to you?

File: test/zdtm/static/maps00.c [hunk 2]
```diff
@@ -158,7 +158,8 @@ static int check_map(struct map *map)
...
+			__builtin___clear_cache(map->ptr, map->ptr + ONE_MAP_SIZE);
```
Comment (avagin) [inline test/zdtm/static/maps00.c:162]: pls add a comment why we need to cleare cache here

Comment (time-river) [reply test/zdtm/static/maps00.c:167]: done

**Comment by time-river** [comment]
> CI is happy. The description sounds reasonable. I have never seen this error but I am not running CRIU very often on aarch64.

This error looks like raise easily in some environment. I ran the case in three environment, only one raised this problem frequently.
